spec/spec.md

```diff
@@ -179,6 +179,19 @@ to the number of positions in the tuple type `T`, an error is reported.
 Given type `tuple(T[0], .., T[n])` for `T`, the type of the expression is
 `T[index]`.
 
+#### Expression Lists
+
+```grammar
+expr += (Exprs <expr>+)
+```
+
+A non-empty list of expressions, where the tail expression may be any type and
+preceding ones must be `unit` or `void`. An error is reported if a a non-tail
+expression is not of `unit` or `void` type.
+
+The type of the expression list is inferred as `void` if any non-trailing
+expression is `void`, otherwise the type is that of the trailing expression.
```

Comment:
A realization on my part/small tidbit:

In Haskell an expression/routine type is not a Set, but instead a Hask, which is a Set + a bottom indicating it might never return. We're headed down a similar road, where `void` sub-types everything means the same. Also, in case anyone is wondering I did think about the implication of dropping this, but there are very few instances where you'd want to ensure an expression is guaranteed to fully evaluate. Even then many of those can be handled through other means, like ensuring all errors are handled (read: no cause for abnormal termination).